### PR TITLE
limiting max bls keys per node to 4 and some node.sh enhancements

### PR DESCRIPTION
**Pull request by gupadhyaya**
* Limits max bls keys per node to 4
* adds feature to read passphrase for bls keys in node.sh
* adds feature to pass bls key folder as input to harmony binary in node.sh

### CODE REVIEW

File: scripts/node.sh [hunk 1]
```diff
@@ -736,6 +746,19 @@ if ! ${multi_key}; then
...
+   for f in ${blsfolder}/*.key 
+   do
+      if [ ! -f $f ]; then
+         err 10 "could not find bls key file: $f"
+      fi
+      passfile=${blsfolder}/$(basename "${f%.*}").pass
+      if [ ! -f "$passfile" ]; then
+         unset -v passphrase
+         read -rsp "Enter passphrase for the BLS key file $f: " passphrase
+         echo ${passphrase} | tee $passfile
```
Comment (LeoHChen) [inline scripts/node.sh]: this is very risky to save the passphrase to a local file w/o user's awareness.  You need to WARN user about this.  Ask user to confirm. Also, think about removing the file after the node is started.

File: cmd/harmony/main.go [hunk 2]
```diff
@@ -273,6 +273,9 @@ func setupStakingNodeAccount() error {
 	return nil
 }
 
+// MaxNumBLSKeysPerNode specifies the maximum number of bls keys that a node can run
+const MaxNumBLSKeysPerNode = 4
```
Comment (LeoHChen) [inline cmd/harmony/main.go]: don't hardcode the number here.  This configuration variable could be network specific, we may use different number for testnet/devnet, etc.  So, I think you should add it as a network configuration parameter.

Comment (gupadhyaya) [reply cmd/harmony/main.go]: you mean in `internal/configs/sharding` or `internal/params/config.go` ?

File: scripts/node.sh (hunk 1 above)
Comment (gupadhyaya) [reply scripts/node.sh]: @LeoHChen added notification for user. Is this sufficient? To implement automatic delete of these files, I need to do that inside harmony program, which is not good. Any other way to do it as part of node.sh?

File: cmd/harmony/main.go (hunk 2 above)
Comment (gupadhyaya) [reply cmd/harmony/main.go]: @LeoHChen ?

Comment (rlan35) [reply cmd/harmony/main.go]: as in a main.go flag.

File: scripts/node.sh (hunk 1 above)
Comment (LeoHChen) [reply scripts/node.sh]: add a parallel background process to delete the file after started harmony program.

Comment (gupadhyaya) [reply scripts/node.sh]: done.

File: scripts/node.sh [hunk 3]
```diff
@@ -804,6 +846,7 @@ do
    *) ld_path_var=LD_LIBRARY_PATH;;
    esac
    run() {
+      (sleep 10 && rm_bls_pass)&
```
Comment (LeoHChen) [inline scripts/node.sh]: use a variable to replace 10 here, I suggest default to 30 sec then.

File: scripts/node.sh [hunk 4]
```diff
@@ -700,6 +710,31 @@ kill_node() {
...
+      if [ ! -f "$passfile" ]; then
+         unset -v passphrase
+         read -rsp "Enter passphrase for the BLS key file $f: " passphrase
+         echo ${passphrase} | tee $passfile
+         echo "Passphrase is temporarely saved to: $passfile, which will be deleted after the node is launched successfully!!!"
+      fi
+   done
+}
+
+rm_bls_pass() {
```
Comment (LeoHChen) [inline scripts/node.sh]: we can give an option to users that they don't have to remove the passfile if they choose to. then we just give them a warning then.

File: scripts/node.sh [hunk 5]
```diff
@@ -812,5 +855,9 @@ do
    esac || msg "node process finished with status $?"
    ${loop} || break
    msg "restarting in 10s..."
+   rm_bls_pass
    sleep 10
+   if ${multi_key}; then
```
Comment (LeoHChen) [inline scripts/node.sh]: this won't work, the program will run in line 853 and 854.

Comment (gupadhyaya) [reply scripts/node.sh]: this is only for the failure case. non failure case is already considered.

File: scripts/node.sh (hunk 4 above)
Comment (gupadhyaya) [reply scripts/node.sh]: done.

**Pull request merged**